backend/danswer/connectors/freshdesk/connector.py

```diff
@@ -0,0 +1,144 @@
+import requests
+import json
+from datetime import datetime, timezone
+from typing import Any, List, Optional
+from danswer.file_processing.html_utils import parse_html_page_basic
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput, PollConnector, LoadConnector, SecondsSinceUnixEpoch
+from danswer.connectors.models import ConnectorMissingCredentialError, Document, Section
+from danswer.utils.logger import setup_logger
+
+logger = setup_logger()
+
+
+class FreshdeskConnector(PollConnector, LoadConnector):
+    def __init__(self, batch_size: int = INDEX_BATCH_SIZE) -> None:
+        self.batch_size = batch_size
+
+    def ticket_link(self, tid: int) -> str:
+        return f"https://{self.domain}.freshdesk.com/helpdesk/tickets/{tid}"
+
+    def build_doc_sections_from_ticket(self, ticket: dict) -> List[Section]:
+        # Use list comprehension for building sections
+        return [
+            Section(
+                link=self.ticket_link(int(ticket["id"])),
+                text=json.dumps({
+                    key: value
+                    for key, value in ticket.items()
+                    if isinstance(value, str)
+                }, default=str),
+            )
+        ]
+
+    def strip_html_tags(self, html: str) -> str:
+        return parse_html_page_basic(html)
+
+    def load_credentials(self, credentials: dict[str, Any]) -> Optional[dict[str, Any]]:
+        self.api_key = credentials.get("freshdesk_api_key")
+        self.domain = credentials.get("freshdesk_domain")
+        self.password = credentials.get("freshdesk_password")
+        return None
+    
+    def _fetch_tickets(self, start: datetime, end: datetime) -> List[dict]:
+        if any([self.api_key, self.domain, self.password]) is None:
```

Comment:
make this function accept datetime | None 

change the filter in the url based on if it recieves a none or an actual datetime value